### DISCUSSION TITLE
License terms for older versions of Yolo implemented by third-party

### DISCUSSION THREAD

**Discussion by deepakgh** [ultralytics/ultralytics#14008] [General]
Please confirm/comment:

The darknet implementation of Yolo (v1,v2,v3, yolo9000) is permissive and can be used for commercial purpose (https://github.com/pjreddie/darknet).

What about the license terms of Yolov7 implementations under Tensorflow Model Garden (https://github.com/tensorflow/models/tree/master/official), which are released under Apache 2.0 License?

Do they need Enterprise License from Ultralytics for use in proprietary, commercial applications?

What about any other third-party implementation of any Yolo models? Do they need Ultralytics Enterprise/commercial license?

**Comment by glenn-jocher** [comment]
@deepakgh,

Thank you for your detailed questions regarding the licensing terms for various YOLO implementations.

1. **Darknet Implementation (YOLOv1, v2, v3, YOLO9000)**:
   - The original YOLO versions implemented in Darknet by Joseph Redmon are under the MIT License, which allows for commercial use without restrictions. Further details can be found in the [Darknet repository](https://github.com/pjreddie/darknet).

2. **YOLOv7**:
   - The YOLOv7 model, available in the repository maintained by WongKinYiu, is licensed under GPL-3.0. This means that any derivative works must also be distributed under GPL-3.0, regardless of whether they originated from a different licensing environment. It’s important to ensure compliance with GPL-3.0 when using or modifying YOLOv7, including any requirements to disclose source code. You can view the GPL-3.0 license and its specifics on the [YOLOv7 GitHub page](https://github.com/WongKinYiu/yolov7).

3. **Other Third-Party Implementations**:
   - For third-party implementations of YOLO models, the specific license will depend on the implementation. However, any modifications or derived works from YOLOv7 specifically must adhere to GPL-3.0 due to its copyleft nature.
   - If using Ultralytics' models, architectures, or code in R&D, development, or deployment stages, and your project is not fully open-sourced under AGPL-3.0, an Ultralytics Enterprise License is necessary for both internal and external commercial use.

For comprehensive licensing information or to apply for an Enterprise License, please visit [Ultralytics Licensing](https://ultralytics.com/license).

Please let me know if you need further assistance or clarification!

**Reply by deepakgh** [reply]
Thanks for the detailed and quick response. 
So, Any third-party implementation of any Yolo models (v0-v8) (e.g., by YoloV8 implementation  available/published on Kaggle or  on github etc.), would be governed by the specific license under which they are released. In other words, the model (at a conceptual level) irrespective of the version, doesn't have any license restrictions but the implementation may have. Is this understanding/interpretation correct? 

I have sent an enquiry to ultralytics. I will follow it up.

**Reply by glenn-jocher** [reply]
@deepakgh you're welcome! Yes, your understanding is correct. The licensing terms apply to the specific implementation of the YOLO models, not the conceptual model itself. Therefore, any third-party implementation of YOLO (v0-v8) will be governed by the specific license under which it is released. For example, a YOLOv8 implementation on Kaggle or GitHub will follow the license specified by the author of that implementation.

If you have any further questions or need additional clarification, feel free to ask. Also, thank you for reaching out to Ultralytics for your inquiry. They will be able to provide detailed information regarding licensing for commercial use.

Best of luck with your projects! 😊

**Reply by deepakgh** [reply]
Thanks! that explains everything.

**Reply by glenn-jocher** [reply]
You're welcome! 😊 I'm glad the explanation was helpful.

If you encounter any issues or have further questions, please don't hesitate to provide a reproducible example. This helps us diagnose and resolve issues more efficiently. You can find guidance on creating a minimum reproducible example here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

Additionally, please ensure you are using the latest versions of the packages, as updates often include important bug fixes and improvements.

Feel free to reach out if you need more assistance. Happy coding! 🚀
